frontendUiRootImpl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/providerservice/VcsRootComboBoxProvider.java

```diff
@@ -0,0 +1,24 @@
+package com.virtuslab.gitmachete.frontend.ui.impl.root.providerservice;
+
+import com.intellij.openapi.components.Service;
+import com.intellij.openapi.project.Project;
+import git4idea.repo.GitRepository;
+import io.vavr.control.Option;
+import lombok.Getter;
+
+import com.virtuslab.gitmachete.frontend.ui.impl.root.VcsRootComboBox;
+
+@Service
+public final class VcsRootComboBoxProvider {
+
+  @Getter
+  private final VcsRootComboBox vcsRootComboBox;
+
+  public VcsRootComboBoxProvider(Project project) {
+    this.vcsRootComboBox = new VcsRootComboBox(project);
```

Comment:
Possibly using `instantiateSoleImplementingClass` for `IGitRepositorySelectionProvider` would be a solution here? and then `providerservice` could be moved to `frontendUiApi`